src/sage/schemes/curves/zariski_vankampen.py

```diff
@@ -243,56 +272,192 @@ def corrected_voronoi_diagram(points):
+         A vertex at (2, -2),
+         A vertex at (2, 2)]
+        sage: cir = E.eulerian_circuit()
+        sage: cir
+        [(A vertex at (-2, -2), A vertex at (2, -2), None),
+         (A vertex at (2, -2), A vertex at (2, 2), None),
+         (A vertex at (2, 2), A vertex at (-2, 2), None),
+         (A vertex at (-2, 2), A vertex at (-2, -2), None)]
+        sage: cir_oriented = orient_circuit(cir); cir_oriented
+        (A vertex at (-2, -2), A vertex at (2, -2), A vertex at (2, 2),
+         A vertex at (-2, 2), A vertex at (-2, -2))
+        sage: cirinv = list(reversed([(c[1],c[0],c[2]) for c in cir]))
+        sage: cirinv
+        [(A vertex at (-2, -2), A vertex at (-2, 2), None),
+         (A vertex at (-2, 2), A vertex at (2, 2), None),
+         (A vertex at (2, 2), A vertex at (2, -2), None),
+         (A vertex at (2, -2), A vertex at (-2, -2), None)]
+        sage: orient_circuit(cirinv) == cir_oriented
+        True
+        sage: cir_oriented == orient_circuit(cir, convex=True)
+        True
+        sage: P0=[(1,1/2),(0,1),(1,1)]; P1=[(0,3/2),(-1,0)]
+        sage: Q=Polyhedron(P0).vertices()
+        sage: Q = [Q[2], Q[0], Q[1]] + [_ for _ in reversed(Polyhedron(P1).vertices())]
+        sage: Q
+        [A vertex at (1, 1/2), A vertex at (0, 1), A vertex at (1, 1),
+         A vertex at (0, 3/2), A vertex at (-1, 0)]
+        sage: E = Graph()
+        sage: for v, w in zip(Q, Q[1:] + [Q[0]]):
+        ....:   E.add_edge((v, w))
+        sage: cir = orient_circuit(E.eulerian_circuit(), precision=1, verbose=True)
+        2
+        sage: cir
+        (A vertex at (1, 1/2), A vertex at (0, 1), A vertex at (1, 1),
+         A vertex at (0, 3/2), A vertex at (-1, 0), A vertex at (1, 1/2))
     """
-    Return the bounded segments of the Voronoi diagram of the given points.
+    vectors = [v[1].vector() - v[0].vector() for v in circuit]
+    circuit_vertex = (circuit[0][0],) + tuple(e[1] for e in circuit)
+    circuit_vertex = tuple(circuit_vertex)
+    if convex:
+        pr = matrix([vectors[0], vectors[1]]).determinant()
+        if pr > 0:
+            # return circuit
+            return circuit_vertex
+        elif pr < 0:
+            # return list(reversed([(c[1], c[0]) + c[2:] for c in circuit]))
+            return tuple(reversed(circuit_vertex))
+    prec = precision
+    while True:
+        CIF = ComplexIntervalField(prec)
+        totalangle = sum((CIF(*vectors[i]) / CIF(*vectors[i - 1])).argument()
+                         for i in range(len(vectors)))
+        if totalangle < 0:
+            # return list(reversed([(c[1], c[0]) + c[2:] for c in circuit]))
+            return tuple(reversed(circuit_vertex))
+        if totalangle > 0:
+            # return circuit
+            return circuit_vertex
+        prec *= 2
+        if verbose:
+            print(prec)
+
+
+def voronoi_cells(V):
+    r"""
+    Compute the graph, the boundary graph, a base point, a positive orientation of the boundary graph,
```

Comment:
80 columns mode

Reply:
Done here and elsewhere. There are still some longer lines in doctests and code.